auth/client/src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -659,6 +664,24 @@ public AuthenticationConfiguration usePassword(Password password) {
         return password == null ? useCredentials(filtered) : useCredentials(filtered).useCredential(new PasswordCredential(password));
     }
 
+    /**
+     * Create a new configuration which is the same as this configuration, but converts the given masked password to a
+     * clear password and uses the clear password to authenticate.
+     *
+     * @param password the password to use
+     * @return the new configuration
+     */
+    public AuthenticationConfiguration useMaskedPassword(MaskedPassword password) throws NoSuchAlgorithmException, InvalidKeySpecException {
+        final PasswordFactory passwordFactory = PasswordFactory.getInstance(password.getAlgorithm());
```

Comment:
It would be good to assert that the password is not null at the beginning here

Reply:
Fixed